extensions/ql-vscode/src/queries-panel/queries-module.ts

```diff
@@ -6,16 +6,42 @@ import { DisposableObject } from "../common/disposable-object";
 import { QueriesPanel } from "./queries-panel";
 import { QueryDiscovery } from "./query-discovery";
 import { QueryPackDiscovery } from "./query-pack-discovery";
+import { QueriesPanelCommands } from "../common/commands";
 
 export class QueriesModule extends DisposableObject {
+  private queriesPanel: QueriesPanel | undefined;
+
   private constructor(readonly app: App) {
     super();
   }
 
+  public static initialize(
+    app: App,
+    cliServer: CodeQLCliServer,
+  ): QueriesModule {
+    const queriesModule = new QueriesModule(app);
+    app.subscriptions.push(queriesModule);
+
+    queriesModule.initialize(app, cliServer);
+    return queriesModule;
+  }
+
+  public getCommands(): QueriesPanelCommands {
+    if (!this.shouldInitializeQueriesPanel) {
+      return {} as any as QueriesPanelCommands;
+    }
+
+    if (!this.queriesPanel) {
+      throw new Error("Queries panel not initialized");
+    }
+
+    return {
+      ...this.queriesPanel.getCommands(),
+    };
+  }
+
   private initialize(app: App, cliServer: CodeQLCliServer): void {
-    if (!isCanary() || !showQueriesPanel()) {
-      // Currently, we only want to expose the new panel when we are in canary mode
-      // and the user has enabled the "Show queries panel" flag.
+    if (!this.shouldInitializeQueriesPanel) {
```

Comment:
This is not calling the method, but is checking that the method exists. It should be (also in the other instances):

```suggestion
    if (!this.shouldInitializeQueriesPanel()) {
```

You can also change the `shouldInitializeQueriesPanel` to be a getter, which would be my preferred solution in this case since there are no real side-effects.

Reply:
Thanks for catching this.